### ISSUE TITLE
everything stopped working in the past few days - no camera connection, no security connection

### ISSUE THREAD

**Issue by eoddc5** [dgreif/ring#1313]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe The Bug

Nothing changed on my end. This either started happening with 12.0.0 beta 4 or 12.0.1 

<img width="690" alt="image" src="https://github.com/dgreif/ring/assets/10222895/eccabdc8-569d-4371-9aab-8461cf5b008f">

I tried to generate a new token, to maybe refresh my login and kick it back into gear - but I keep getting this error after trying to login: 

<img width="284" alt="image" src="https://github.com/dgreif/ring/assets/10222895/7fb01634-0e23-4567-987c-190606ee01f2">

@dgreif : I can do whatever you want to debug or troubleshoot, if you let me know

### Expected behavior

proper connection to devices

### Homebridge Ring Config

```shell
"refreshToken": "sanitized=",
            "unbridgeCameras": true,
            "hideDoorbellSwitch": true,
            "hideCameraSirenSwitch": true,
            "hideInHomeDoorbellSwitch": true,
            "hideAlarmSirenSwitch": true,
            "_bridge": {
                "username": "sanitized",
                "port": 43919
            },
            "platform": "Ring"
```

### OS

macOS

### Node.js Version

18.17.1

### NPM Version

8.x.x

### Homebridge/HOOBs Version

1.6.1

### Homebridge Ring Plugin Version

12.0.1

### Operating System

macOS

**Comment by eoddc5** [comment]
> If you token was expired for a period of time it's possible your IP address has been blocked. I would suggest disabling the plugin and waiting 8-12 hours, then re-enable and try authenticating again. I'm not 100% how long it takes to unblock, but the block will stay as long as there are ongoing failed authentication attempts. There are no currently no known widespread issues with authentication (I just verified that Ring authentication in the plugin is working fine for me).

awesome

ill give that a try!

thanks

**Comment by lgardn3** [comment]
I'm having a similar issue the past few days on my Synology NAS. Have a bunch of child bridges so it's not ruing everything, but very unresponsive with ring. 

Failed to reach Ring server at https://api.ring.com/clients_api/ring_devices.  Timeout awaiting 'request' for 20000ms.  Trying again in 5 seconds...

**Comment by eoddc5** [comment]
Yeah mine is on a bridge so it’s isolated. @tsightler one thing that I’m unsure of about the above comment from you is that:

I have 14 total ring devices (counting window and door sensors)

only the ring cameras and the security station stopped responding

All the sensors continued to show responsive. Now with the whole thing disabled they’re all showing unresponsive (which yes makes sense)

**Comment by lgardn3** [comment]
^Same issue. When the cameras and alarm aren't working, the sensors still seem to be responding

**Comment by tsightler** [comment]
So far nobody has posted any debug logs, so there's not much to go on here. @eoddc5 My guess was based only on the fact that you can't re-authenticate because there's nothing else to go on.  I assumed you were re-authenticating because you were getting invalid token errors as that's the only real reason to do so, but without any logs the authentication error was the only thing to go on.

**Comment by jarfgames** [comment]
Not sure if this is the same issue, but noticed all of my Ring devices disappeared out of HomeKit the past few days.... thought I'd try to re-install 12.0.1, but it's erroring out. Same issue if I try to install an earlier version or uninstall the plugin. Here is the log:

0 verbose cli /usr/bin/node /usr/bin/npm
1 info using npm@9.6.7
2 info using node@v18.17.1
3 timing npm:load:whichnode Completed in 7ms
4 timing config:load:defaults Completed in 10ms
5 timing config:load:file:/usr/lib/node_modules/npm/npmrc Completed in 1ms
6 timing config:load:builtin Completed in 3ms
7 timing config:load:cli Completed in 15ms
8 warn config global-style This option has been deprecated in favor of `--install-strategy=shallow`
9 timing config:load:env Completed in 5ms
10 timing config:load:project Completed in 4ms
11 timing config:load:file:/home/pi/.npmrc Completed in 0ms
12 timing config:load:user Completed in 2ms
13 timing config:load:file:/usr/etc/npmrc Completed in 1ms
14 timing config:load:global Completed in 1ms
15 timing config:load:setEnvs Completed in 6ms
16 timing config:load Completed in 48ms
17 timing npm:load:configload Completed in 49ms
18 timing npm:load:mkdirpcache Completed in 1ms
19 timing npm:load:mkdirplogs Completed in 0ms
20 verbose title npm uninstall homebridge-ring
21 verbose argv "uninstall" "--global" "homebridge-ring"
22 timing npm:load:setTitle Completed in 7ms
23 timing config:load:flatten Completed in 16ms
24 timing npm:load:display Completed in 21ms
25 verbose logfile logs-max:10 dir:/home/pi/.npm/_logs/2023-09-22T03_19_10_800Z-
26 verbose logfile /home/pi/.npm/_logs/2023-09-22T03_19_10_800Z-debug-0.log
27 timing npm:load:logFile Completed in 50ms
28 timing npm:load:timers Completed in 0ms
29 timing npm:load:configScope Completed in 0ms
30 timing npm:load Completed in 140ms
31 timing config:load:flatten Completed in 7ms
32 timing arborist:ctor Completed in 4ms
33 timing idealTree:init Completed in 33ms
34 timing idealTree:userRequests Completed in 3ms
35 silly idealTree buildDeps
36 timing idealTree:#root Completed in 1ms
37 timing idealTree:buildDeps Completed in 8ms
38 timing idealTree:fixDepFlags Completed in 2ms
39 timing idealTree Completed in 51ms
40 silly logfile start cleaning logs, removing 1 files
41 silly logfile done cleaning log files
42 timing reify:loadTrees Completed in 1042ms
43 timing reify:diffTrees Completed in 33ms
44 silly reify mark retired [ '/usr/lib/node_modules/homebridge-ring' ]
45 silly reify moves {
45 silly reify   '/usr/lib/node_modules/homebridge-ring': '/usr/lib/node_modules/.homebridge-ring-J83NaDAa'
45 silly reify }
46 timing reify:rollback:retireShallow Completed in 1ms
47 timing command:uninstall Completed in 1158ms
48 verbose stack Error: ENOTEMPTY: directory not empty, rename '/usr/lib/node_modules/homebridge-ring' -> '/usr/lib/node_modules/.homebridge-ring-J83NaDAa'
49 verbose cwd /usr
50 verbose Linux 5.10.103-v7+
51 verbose node v18.17.1
52 verbose npm  v9.6.7
53 error code ENOTEMPTY
54 error syscall rename
55 error path /usr/lib/node_modules/homebridge-ring
56 error dest /usr/lib/node_modules/.homebridge-ring-J83NaDAa
57 error errno -39
58 error ENOTEMPTY: directory not empty, rename '/usr/lib/node_modules/homebridge-ring' -> '/usr/lib/node_modules/.homebridge-ring-J83NaDAa'
59 verbose exit -39
60 timing npm Completed in 2980ms
61 verbose unfinished npm timer reify 1695352752622
62 verbose unfinished npm timer reify:retireShallow 1695352753729
63 verbose code -39
64 error A complete log of this run can be found in: /home/pi/.npm/_logs/2023-09-22T03_19_10_800Z-debug-0.log

**Comment by tsightler** [comment]
That's an NPM issue, you should remove the directory, probably needs sudo.

**Comment by jarfgames** [comment]
> That's an NPM issue, you should remove the directory which it says is not empty, probably a filesystem permissions issues, maybe needs sudo.

Thanks for the quick response.. so in my case, would that be:

sudo rm -r /usr/lib/node_modules/.homebridge-ring-J83NaDAa

**Comment by tsightler** [comment]
Seems like it.  It's a little difficult to be 100% sure since there are so many different ways to install HB, but removing that temporary directory shouldn't be an issue.  However, permissions may still be a problem.

**Comment by eoddc5** [comment]
well I feel like an ass now. I figured it out. that along with my Scrypted I just was checking and I couldn't access it

my VPN updated itself I guess and was auto connecting on start up. something I NEVER allow. so I didn't notice my vpn was connecting - thus, breaking these things.

turned it off, immediately reconnected

this might help others !

**Comment by jarfgames** [comment]
> Seems like it. It's a little difficult to be 100% sure since there are so many different ways to install HB, but removing that temporary directory shouldn't be an issue. However, permissions may still be a problem.

Had to do both of the following, and then a re-install of the plugin:

sudo rm -r /usr/lib/node_modules/.homebridge-ring-J83NaDAa
sudo rm -r /usr/lib/node_modules/homebridge-ring

Lost all of my HomeKit set up (room allocations, automations) related to Ring devices, but everything is working again. Thanks for your help.
